### ISSUE TITLE
DataTable not drawn when wrapped inside a widget (Static)

### ISSUE THREAD

**Issue by Riyum** [Textualize/textual#2230]
Hello !
I noticed that after the 0.17.0 update, the following code will not produce a DataTable.
the issue appears to be caused by calling `_update` in `on_mount`, as the problem is resolved when that line is commented out.

```python
from textual.app import App, ComposeResult
from textual.widgets import DataTable, Static
from textual.screen import Screen


class MyTable(Static):
    def compose(self) -> ComposeResult:
        yield DataTable()

    def on_mount(self) -> None:
        self._update()
        self.set_interval(1, self._update)
        # self.set_timer(0.3, self.forceResize)

    def _update(self) -> None:
        dt = self.query_one(DataTable)
        dt.clear(True)
        dt.add_columns("N", "N*10", "N*100", "N*1000")
        for n in range(20):
            dt.add_row(str(n), str(n * 10), str(n * 100), str(n * 1000))

    def forceResize(self) -> None:
        dt = self.query_one(DataTable)
        dt.styles.display = "none"
        dt.styles.display = "block"


class MyScreen(Screen):
    def compose(self) -> ComposeResult:
        yield MyTable()


class Test(App):

    CSS = """
    MyScreen{
        border: solid red;
    }
    MyTable {
        border: solid yellow;
        width: 100%;
        height: 100%;
    }
    DataTable {
        border: solid blue;
    }
    """

    SCREENS = {"ms": MyScreen()}

    def on_mount(self) -> None:
        self.push_screen("ms")


Test().run()
```

some notes:
- In case I want to call `_update` in `on_mount`, I must also call (with a timer) `forceResize` for the table to become visible.
- If I move all the code from `MyTable` to `MyScreen` the issue is gone

**Comment by barnybug** [comment]
I've been bitten by the same issue which seems to have appeared when you say in 0.17.0. My test case is very similar:
```
from textual.app import App, ComposeResult
from textual.widgets import (
    DataTable,
    Label,
    Static,
)


class Journal(Static):
    def compose(self) -> ComposeResult:
        yield DataTable()

    def on_mount(self):
        table = self.query_one(DataTable)
        table.add_columns("Column 1", "Column 2")
        table.add_rows([["A", "1"]])

class TableApp(App):
    CSS = """
#journal {
    height: 1fr;
}
"""

    def compose(self) -> ComposeResult:
        yield Journal(id="journal")


if __name__ == "__main__":
    TableApp().run()
```

**Comment by FuG** [comment]
As a temporary quick fix, I have found that changing the font size of the terminal/console makes it show up again.
(FYI - I am using WezTerm on Win10).

**Comment by willmcgugan** [comment]
We will take a look at this. In the meantime, adding `table.refresh(layout=True)` may fix it.

**Comment by willmcgugan** [comment]
Have you tried `0.19.1` ?

**Comment by Riyum** [comment]
> Have you tried `0.19.1` ?

Nope, I was on `0.19.0`.
version `0.19.1` did indeed fix the issue without any needed workarounds :ok_hand:

**Comment by Riyum** [comment]
fixed by 4f7b2d00bc0429711edf223534486136189b2d70
